port-chooser/src/main/java/org/terracotta/utilities/test/runtime/Os.java

```diff
@@ -0,0 +1,864 @@
+/*
+ * Copyright 2020 Terracotta, Inc., a Software AG company.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.utilities.test.runtime;
+
+import org.terracotta.utilities.test.io.WindowsSpecialFolder;
+
+import java.io.IOException;
+import java.util.Locale;
+
+/**
+ * O/S functions
+ *
+ * @author teck
+ */
+// Migrated from https://github.com/Terracotta-OSS/terracotta-core/blob/master/common/src/main/java/com/tc/util/runtime/Os.java
+public class Os {
```

Comment:
This really feels like it should be an enum. Which might itself lead to more changes.

Reply:
I don't disagree but I picked this up solely for use in `EphemeralPorts`.  Is it worth the effort?

Reply:
Maybe not... the cascading enum-ification might lead to some code cleanup and motion between `EphemeralPorts` and here but it's unlikely to be a profound set of changes.